### PR TITLE
ENH: remove fixups

### LINKED ISSUE #1821: remove spits outs an error (but complets the act)
```shell
(git-annex)hopa:/tmp/ds123-clone[master]git-annex
$> datalad remove 123
drop(ok): /tmp/ds123-clone/123 (file)
remove(ok): 123
[ERROR  ] 'NoneType' object has no attribute 'endswith' [posixpath.py:join:70] (AttributeError) 
(dev) 2 11128 ->1.....................................:Mon 11 Sep 2017 03:47:09 PM EDT:.
(git-annex)hopa:/tmp/ds123-clone[master]git-annex
$> ls -l 123
ls: cannot access '123': No such file or directory
```

edit 1: actually doesn't complete the act -- fails before saving... fixing it

### LINKED ISSUE #1822: we need --nosave to remove as well
as discovered while playing with #1821-- for consistent (with `add`) API / expected behaviour

### PR DESCRIPTION
**Pull request by yarikoptic**
- ENH: Added  --nosave and --message to  remove cmd   Closes #1822
- BF: make remove not fail if operated without a specified dataset  Closes #1821

### CODE REVIEW

**Comment by yarikoptic** [comment]
Any concerns @datalad/developers ? Or let's get it merged

File: datalad/distribution/remove.py [hunk 1]
```diff
@@ -270,15 +276,18 @@ def __call__(
...
                 # TODO compose hand-selected annotated paths
                 path=to_save,
                 # we might have removed the reference dataset by now, recheck
-                dataset=refds_path if GitRepo.is_valid_repo(refds_path) else None,
-                # TODO allow for custom message
-                #message=message if message else '[DATALAD] removed content',
-                message='[DATALAD] removed content',
+                dataset=refds_path
+                        if (refds_path and GitRepo.is_valid_repo(refds_path))
+                        else None,
```
Comment (bpoldrack) [inline datalad/distribution/remove.py]: To solve it that way doesn't really make sense to me. Judging from #1821 the determination of the reference datasets fails for some reason and needs to be fixed. It's not about having none. You can't `datalad remove` anything from a non-repository, right?

File: datalad/distribution/remove.py [hunk 2]
```diff
@@ -94,6 +98,8 @@ def __call__(
             dataset=None,
             recursive=False,
             check=True,
+            save=True,
+            message=None,
```
Comment (bpoldrack) [inline datalad/distribution/remove.py]: I think it's inconsistent to have `save`/`nosave` (and consequently `message`) in `remove` but not in `add`.
Those are not meant to be commands that leave some staged things behind. ATM, I feel this undermines the API concept.

File: datalad/distribution/remove.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/distribution/remove.py]: well... I am not strong in all this logic... looking at original code though it even allowed refds_path point to some invalid repo as far as I see it ;)  so checking if it is defined at all, sounds legit to me.  moreover we could fix up instead is_valid_repo to not puke upon none?

Comment (yarikoptic) [reply datalad/distribution/remove.py]: BTW, I did add a test for this usecase, so if you see a better/proper fix, you could check it out quickly.

File: datalad/distribution/remove.py (hunk 2 above)
Comment (yarikoptic) [reply datalad/distribution/remove.py]: #1829 could probably the venue to discuss this further.  Also ref https://github.com/datalad/datalad/issues/1822#issuecomment-329766900 
meanwhile this one fixes an issue of remove crashing and save/nosave is already present in other command(s), so imho it is consistent in general. indeed `add` should get the same powers, and vise versa.  Also, in principle, all those which save should acquire all the powers of save (e.g. `-S`)

Comment (bpoldrack) [reply datalad/distribution/remove.py]: Actually, `add` has it already. I was wrong about that.

File: datalad/distribution/remove.py (hunk 1 above)
Comment (bpoldrack) [reply datalad/distribution/remove.py]: Either I have misread the diff or sth else was merged in the meantime. Whatever is the case - what I was unhappy about isn't valid (anymore).

**Pull request merged**